htdocs/fourn/commande/card.php

```diff
@@ -400,6 +400,8 @@
 				$localtax1_tx= get_localtax($tva_tx, 1, $mysoc, $object->thirdparty, $tva_npr);
 				$localtax2_tx= get_localtax($tva_tx, 2, $mysoc, $object->thirdparty, $tva_npr);
 
+				$remise_percent = $productsupplier->remise_percent?$productsupplier->remise_percent:$remise_percent;
```

Comment:
If a user enter a line with a discount of 6% in the form to add a new line, we should not use another discount because the supplier has another one because the user will not understand why a 7% discount was saved if he entered 6% on the screen to add a new line.
Instead, i recommand to add javascript so when selecting the product reference, dolibarr set automatically the value of field of discount line in the form. So user see the discount that will be applied and can modify it.